### ISSUE TITLE
TLS handshake connection been reset by envoy after client-hello 

### ISSUE THREAD

**Issue by bincxz** [envoyproxy/envoy#20642]
*Title*: *TLS handshake connection been reset by envoy after client-hello *

*Description*:
>
Hi guys, I have met some really confusing problem. I'm trying to use envoy as a https upstream server proxy.   To achieve this, I have create a listener which listen on  port 10102 to accept https requests. The flowing picture is my envoy config.
<img width="1094" alt="9c6815aed417ce4503bb8359177ba53" src="https://user-images.githubusercontent.com/16399091/161373781-28017562-5926-48dd-9c33-c8ce23b420ca.png">

But when I trying to send a https request to this listener.  the handshake always failed because of the envoy server rst response. It seems that the request didn't hit any filter_chain's match rule.

By the way I'm pretty sure the client's sni option has been set correctly.
![image](https://user-images.githubusercontent.com/16399091/161374042-5e75a306-58c5-4130-956a-697681a645e1.png)

Is there some thing wrong about the usecase?  Thank you very much for your help

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by lambdai** [comment]
1. Can you confirm this filter chain should be selected? If *all* the filter chains in this listener has only "server_names" as the filter chain match, you should be good. Otherwise, you need to read the filter chain document or dump the `filter_chain` name in the access log to confirm. 

2. Have you set downstream tls context for this filter chain? 

3. Have you enable the tls_inspector listener filter? That is the Envoy part to parse the SNI in your request and go through the filter chain matches.

**Comment by lambdai** [comment]
Thanks .
`readacted` is expected. That's how config_dump is supposed to be working. 
Now that this filter chains configures downstream tls context, It's very likely the filter chain is not selected. Otherwise, a tls packet record should be sent back, not a RST

**Comment by bincxz** [comment]
> Thanks . `readacted` is expected. That's how config_dump is supposed to be working. Now that this filter chains configures downstream tls context, It's very likely the filter chain is not selected. Otherwise, a tls packet record should be sent back, not a RST

This problem has been solved.And yes , it caused by not setting the tls_inspector filter.  After add this filter to listener. The handshake established successfully.

And there is one small question left,  After I set tls_inspector, the tls worked,  but this filter configuration seems not appeared in config_dump. That is to say, compared to before setting tls_inspector, the config_dump of this Listener does not seem to have changed  (But yes it worked anyway).  Don't know if this is expected. 

![image](https://user-images.githubusercontent.com/16399091/162793204-8ba585e3-b69b-4779-a0c7-e8cfe42e5651.png)


Thanks for your help ~
